feat(gpu): introduce GPU request spec (#1156)
NVIDIA/OpenShell: crates/openshell-driver-kubernetes/README.md

## GPU Support

When `resource_requirements.gpu` is present, the driver checks node allocatable
capacity for `nvidia.com/gpu` and sets the workload's `nvidia.com/gpu` resource

> limit. Requests without an explicit count use one GPU. The sandbox image must

Author: ```suggestion
limit. The default is to request a single GPU. The sandbox image must
```

Author: Done. The Kubernetes README now states that the default is to request a single GPU.
